### ISSUE TITLE
XRef: Tab-to-autocomplete in IPython + Python3 results in Spark job execution

### ISSUE THREAD

**Issue by parente** [ipython/ipython#10580]
Opening an issue here just to raise awareness across communities for now.

https://issues.apache.org/jira/browse/SPARK-20818

**Comment by Carreau** [comment]
Sight. Thanks for cross-linking. Deactivating Jedi should work. There is an
option for that.

**Comment by parente** [comment]
I was thinking it might be Jedi but I want to double check that it also wasn't happening in ipython 5.x without Jedi.

**Comment by Carreau** [comment]
Yeah, it is Jedi. I've found the line that does that. I'm unsure how to
avoid the bug though. I'm on objects that define custom __get__ there is
another issue open about that I believe.

**Comment by parente** [comment]
I see the other issues:

* #10521
* davidhalter/jedi#925

In the Spark case, it appeared to only have the side effect when getting possible completions for methods wrapped with the Python 3 decorator that modifies the __doc__ of methods.

**Comment by parente** [comment]
Notebook that reproduces the bug using one of the docker-stacks images: https://gist.github.com/parente/1a1bef370148a3e5e3234af1091f969d.

Disabling jedi next just to make sure it clears up the issue (though I'm almost certain it will).

**Comment by parente** [comment]
Confirmed: disabling jedi avoids the problem.
